com.microsoft.mrtk.tools/SubsystemWizard/SubsystemGenerator.cs

```diff
@@ -0,0 +1,605 @@
+        private const string DescriptorTemplateGuid = "8e1afb29fbaf1c2419511d266f49b976";
+        // SubsystemWizard/Templates/SubsystemInterfaceTemplate.txt
+        private const string InterfaceTemplateGuid = "9bac30e514266984d947e360cfd17b05";
+
+        private const bool DefaultCreateConfiguration = false;
+        private static readonly string DefaultBaseSubsystemName = $"NewSubsystem";
+        private static readonly string DefaultOrganizationName = "Contoso";
+        private static readonly string OutputFolderRoot = Path.Combine("Assets", "MRTK.Generated");
+
+        [SerializeField]
+        private SubsystemWizardState state = SubsystemWizardState.Start;
+
+        /// <summary>
+        /// The current state of the wizard.
+        /// </summary>
+        public SubsystemWizardState State
+        {
+            get => state;
+            set => state = value;
+        }
+
+        [SerializeField]
+        private string baseClassName = DefaultBaseSubsystemName;
+
+        /// <summary>
+        /// The name of the base class from which the subsysten will derive.
+        /// </summary>
+        public string BaseClassName
+        {
+            get => baseClassName;
+            set => baseClassName = value;
+        }
+
+        private bool createConfiguration = DefaultCreateConfiguration;
+
+        /// <summary>
+        /// Inidcates whether or not the wizard should generate a subsystem configuration
+        /// source file.
+        /// </summary>
+        public bool CreateConfiguration
+        {
+            get => createConfiguration;
+            set => createConfiguration = value;
+        }
+
+        [SerializeField]
+        private string organizationName = DefaultOrganizationName;
+
+        /// <summary>
+        /// The name of the entity which is creating / releasing the subsystem.
+        /// </summary>
+        public string OrganizationName
+        {
+            get => organizationName;
+            set => organizationName = value;
+        }
+
+        /// <summary>
+        /// Name of configuration class, if enabled by <see cref="CreateConfiguration"/>
+        /// to create for new subystem.
+        /// </summary>
+        public string ConfigurationName
+        {
+            get => CreateConfiguration ? $"{BaseClassName}Config" : "BaseSubsystemConfig"; 
+        }
+
+        /// <summary>
+        /// Name of descriptor class to create for new subystem.
+        /// </summary>
+        public string DescriptorName
+        {
+            get => $"{BaseClassName}Descriptor";
+        }
+
+        /// <summary>
+        /// The name that will be displayed in project settings.
+        /// </summary>
+        public string DisplayName => $"{OrganizationName} {BaseClassName}";
+
+        /// <summary>
+        /// Name of interface to create for new subystem.
+        /// </summary>
+        public string InterfaceName
+        {
+            get => $"I{BaseClassName}";
+        }
```

Comment:
nit: mixing styles here (and a couple other places) 🙂 I personally prefer the top one!

Reply:
will fix

Reply:
fixed